src/server.ts

```diff
@@ -190,6 +192,10 @@ export default class Server {
 
       if (this.lazyInit) {
         try {
+          /**
+           * Don't await this so that "format on save" is not blocked.
+           * We still need to run the initializers so that we have hinting in component files though.
+           */
           await this.executeInitializers();
```

Comment:
yep, we need to await for initializers here to allow re-write some server logic by addon (if needed)